Utilities/Maintenance/ApplyScriptToRemotes.sh

```diff
@@ -0,0 +1,157 @@
+read -p "password: " -s password
+
+script=$1
+feature_branch=$2
+commit_message=$3
+
+remotes=()
+
+function list_candidate_remotes() {
+  if [ $# -ge 1 ] && [ $1 = "-ongreen" ]; then
+    ongreen=1
+    failing=()
+  else
+    ongreen=0
+  fi
+
+  # Loop over the remote modules' CMake files
+  for filename in ${remote_modules_path}/*.cmake; do
+    echo -n "."
+
+    # Get the current module name
+    module_name=$(grep -v "\s*#" $filename | grep -m 1 -o "itk_fetch_module(\(\w\|\.\|\-\|_\)*")
+    module_name=${module_name/*itk_fetch_module(/}
+
+    # Get the current commit hash
+    curr_commit=$(grep -v "\s*#" $filename | grep -m 1 -o "GIT_TAG \(\w\|\.\|\-\|_\)*")
+    curr_commit=${curr_commit/*GIT_TAG /}
+
+    # Read the git repository information
+    repository=$(grep -v "\s*#" $filename | grep -m 1 -o "GIT_REPOSITORY \${git_protocol}://github.com/\(\w\|\-\|_\|/\)*")
+    repository=${repository/*GIT_REPOSITORY \${git_protocol\}:\/\/github.com\//}
+
+    # Get the latest git commit hash of the remote module.
+    # Remotes will usually not be tagged.
+    latest_commit=$(git ls-remote git://github.com/$repository refs/heads/master)
+    latest_commit=${latest_commit/refs\/heads\/master/}
+
+    # Skip remotes whose current commit in ITK differs from the latest
+    if [ $curr_commit != $latest_commit ]; then
+      continue
+    fi
+
+    # Get the remotes' build status
+    if [ $ongreen -eq 1 ]; then
+      # Get the repository status
+      # Be careful of rate limiting (https://developer.github.com/v3/#rate-limiting)
+      # alternatively using Azure: https://docs.microsoft.com/en-us/azure/data-factory/monitor-programmatically#rest-api
+      status=$(curl -s https://api.github.com/repos/$repository/commits/$latest_commit/status | grep -m 1 -o "\"state\": \"\w*\"")
+      if [ "$status" != "\"state\": \"success\"" ]; then
+        continue
+      fi
+    fi
+
+    remotes+=($module_name)
+
+  done
+}
+
+function configure_itk_remotes() {
+  mkdir ../../../ITK-build
+  cd ../../../ITK-build
+
+  cmake_remote_flags=()
+
+  # Switch on all remotes that have a success build status
+  for remote in ${remotes[@]}; do
+    cmake_remote_flags+=(-D'Module_'$remote':BOOL=ON' )
+  done
+
+  cmake ${cmake_remote_flags[@]} ../ITK
+}
+
+function apply_script_and_push_remotes() {
+  cd ../ITK/Modules/Remote
+
+  # Loop over remotes and apply the changes
+  for remote in ${remotes[@]}; do
+    echo -e $remote
+
+    cd $remote
+
+    repository_basename=$(basename -s .git `git config --get remote.origin.url`)
+
+    git checkout master
+    git checkout -b $feature_branch origin/master
+    $script
+
+    # Add the files, adding filters if necessary
+    git add -u *.txt ./*.cmake ./*.rst ./*.py ./*.yml ./*.c ./*.cxx ./*.h ./*.hxx ./*.wrap
+
+    # Commit and push to the feature branch
+    git commit -m $commit_message
+    git push --quiet https://$username:$password@github.com/$username/$repository_basename $feature_branch
```

Comment:
- Rather that keeping the remote URLs from the `list_candidate_remotes` function, I do get the URL again in `apply_script_and_push_remotes`. I though it was easier than keeping a list from the former method, then finding the appropriate element for the remote at issue (partly because the URL includes the username and password for the push to be done). If you feel there is an alternative to this, please go ahead.